grype/version/rpm_version.go

```diff
@@ -151,10 +151,11 @@ func compareRpmVersions(a, b string) int {
 	// get alpha/numeric segments
 	segsa := alphanumPattern.FindAllString(a, -1)
 	segsb := alphanumPattern.FindAllString(b, -1)
-	segs := int(math.Min(float64(len(segsa)), float64(len(segsb))))
+	maxSegs := int(math.Max(float64(len(segsa)), float64(len(segsb))))
```

Comment:
Why are we casting to a float here? Also, I think this code may pre-date built-in `min` and `max` functions, and should just use them? https://go.dev/ref/spec#Min_and_max

Reply:
TIL! I'll update it 👍